### PR TITLE
bpo-38348: Extend options of ast tool

### PR DESCRIPTION
**Pull request by isidentical**
`--type-comments` and `--indent`

https://bugs.python.org/issue38348

### CODE REVIEW

**Review by brandtbucher — APPROVED** [review]
This looks good!

File: Lib/ast.py [hunk 1]
```diff
@@ -562,15 +562,19 @@ def main():
     parser.add_argument('-m', '--mode', default='exec',
                         choices=('exec', 'single', 'eval', 'func_type'),
                         help='specify what kind of code must be parsed')
+    parser.add_argument('-t', '--type-comments', default='store_true',
```
Comment (ilevkivskyi) [inline Lib/ast.py]: Since the default is `True` wouldn't it be better to have a `--no-type-comments` flag?

Comment (brandtbucher) [reply Lib/ast.py]: Ah, looks like a typo. They probably meant:

```suggestion
    parser.add_argument('-t', '--type-comments', action='store_true',
```

Comment (isidentical) [reply Lib/ast.py]: Agreed

File: Doc/library/ast.rst [hunk 2]
```diff
@@ -368,10 +368,19 @@ The following options are accepted:
    Specify what kind of code must be compiled, like the *mode* argument
    in :func:`parse`.
 
+.. cmdoption:: --no-type-comments
+
+   Don't parse type comments.
```
Comment (vstinner) [inline Doc/library/ast.rst:373]: You have to document that it's new, using ``.. versionadded:: 3.9`` for example (indent it properly, check usage in another documentation file).

File: Doc/library/ast.rst [hunk 3]
```diff
@@ -368,10 +368,19 @@ The following options are accepted:
...
+
 .. cmdoption:: -a, --include-attributes
 
    Include attributes such as line numbers and column offsets.
 
+.. cmdoption:: -i <indent>
+               --indent <indent>
+
+   Indentation of nodes in AST.
```
Comment (vstinner) [inline Doc/library/ast.rst]: Ditto (versionadded).

The indentation is made of whitespaces, no? If yes, I suggest:

```suggestion
   Indentation of nodes in AST (number of spaces).
```

**Review by vstinner — CHANGES_REQUESTED** [review]

File: Doc/library/ast.rst (hunk 2 above)
Comment (isidentical) [reply Doc/library/ast.rst:373]: I dont think we need to specify it, this part of the document already covered by `.. versionadded:: 3.9` https://github.com/python/cpython/commit/832e8640086ac4fa547c055a72929879cc5a963a#diff-37ef1f6c2c9a9c8e36b33fa081e9f5e8R345-R348

**Comment by isidentical** [comment]
I have made the requested changes; please review again

File: Doc/library/ast.rst (hunk 2 above)
Comment (vstinner) [reply Doc/library/ast.rst:373]: Oh, right.

File: Lib/ast.py [hunk 4]
```diff
@@ -562,15 +562,19 @@ def main():
     parser.add_argument('-m', '--mode', default='exec',
                         choices=('exec', 'single', 'eval', 'func_type'),
                         help='specify what kind of code must be parsed')
+    parser.add_argument('--no-type-comments', default=True, action='store_false',
+                        help='dont add information about type comments')
     parser.add_argument('-a', '--include-attributes', action='store_true',
                         help='include attributes such as line numbers and '
                              'column offsets')
+    parser.add_argument('-i', '--indent', type=int, default=3,
+                        help='indent level')
```
Comment (vstinner) [inline Lib/ast.py]: I suggest to reuse the documentation: "indentation of nodes in AST (number of spaces)", or shorter: "indentation of nodes (number of spaces)".

File: Doc/library/ast.rst [hunk 5]
```diff
@@ -368,10 +368,19 @@ The following options are accepted:
...
+   Don't parse type comments.
+
 .. cmdoption:: -a, --include-attributes
 
    Include attributes such as line numbers and column offsets.
 
+.. cmdoption:: -i <indent>
+               --indent <indent>
+
+   Indentation of nodes in AST (number of spaces).
```
Comment (vstinner) [inline Doc/library/ast.rst:382]: I'm not sure that "in AST" is useful. It sounds redundant, no?

File: Misc/NEWS.d/next/Library/2019-10-02-18-15-28.bpo-38348._-5eq2.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+Add indent level and type comments parsing options to ast tool.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2019-10-02-18-15-28.bpo-38348._-5eq2.rst]: Would you mind to mention the option directly? Something like:

```
Add ``-i`` and ``--indent`` (indent level), and ``--no-type-comments``
(type comments) command line options to ast parsing tool.
```

File: Lib/ast.py [hunk 7]
```diff
@@ -562,15 +562,19 @@ def main():
     parser.add_argument('-m', '--mode', default='exec',
                         choices=('exec', 'single', 'eval', 'func_type'),
                         help='specify what kind of code must be parsed')
+    parser.add_argument('--no-type-comments', default=True, action='store_false',
+                        help='dont add information about type comments')
```
Comment (vstinner) [inline Lib/ast.py]:
```suggestion
                        help="don't add information about type comments")
```

File: Misc/NEWS.d/next/Library/2019-10-02-18-15-28.bpo-38348._-5eq2.rst [hunk 8]
```diff
@@ -0,0 +1,2 @@
+Add ``-i`` and ``--indent`` (indent level), and ``--no-type-comments``
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2019-10-02-18-15-28.bpo-38348._-5eq2.rst]:
```suggestion
Add ``-i`` and ``--indent`` (indentation level), and ``--no-type-comments``
```

**Review by vstinner — COMMENTED** [review]
Two more minor requests

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**